orca/orca-spec.md

```diff
@@ -0,0 +1,278 @@
+
+### Binary Protobuf
+
+For Protocol Buffers aware code, this will be a binary serialized base64 encoded OrcaLoadReport
+protobuf in `endpoint-load-metrics` header with BIN prefix:
+
+`endpoint-load-metrics: BIN CZqZmZmZmbk/MQAAAAAAAABAQg4KA2ZvbxGamZmZmZm5P0IOCgNiYXIRmpmZmZmZyT8=`
+
+It could also be specified in the `endpoint-load-metrics-bin` header without BIN prefix:
+
+`endpoint-load-metrics-bin: CZqZmZmZmbk/MQAAAAAAAABAQg4KA2ZvbxGamZmZmZm5P0IOCgNiYXIRmpmZmZmZyT8=`
+
+### Native HTTP
+
+Comma separated key-value pairs in `endpoint-load-metrics`. This is a flattened
+representation of OrcaLoadReport, with the map fields elided into the top level scope by prepending
+the ‘<map_name>.’:
+
+`endpoint-load-metrics: TEXT cpu_utilization=0.3, mem_utilization=0.8, rps_fractional=10.0, eps=1,
+named_metrics.custom_metric_util=0.4`
+
+### JSON
+
+The JSON encoding of OrcaLoadReport:
+
+`endpoint-load-metrics: JSON {“cpu_utilization”: 0.3, “mem_utilization”: 0.8, “rps_fractional”: 10.0,
+“eps”: 1, “named_metrics”: {“custom-metric-util”: 0.4}}`
+
+## Out-of-Band (OOB)
+
+OOB reporting involves an additional load reporting agent that does not sit in the request path.
+This is periodically sampled with sufficient frequency to provide temporal association with
+requests. The data plane LB may either initiate a connection to the agent or offer a gRPC service
+for agents to report to:
+
+```
+// Out-of-band (OOB) load reporting service for the additional load reporting
+// agent that does not sit in the request path. Reports are periodically sampled
+// with sufficient frequency to provide temporal association with requests.
+// OOB reporting compensates the limitation of in-band reporting in revealing
+// costs for backends that do not provide a steady stream of telemetry such as
+// long running stream operations and zero QPS services. This is a server
+// streaming service, client needs to terminate current RPC and initiate
+// a new call to change backend reporting frequency.
+service OpenRcaService {
+  rpc StreamCoreMetrics(OrcaLoadReportRequest) returns (stream xds.data.orca.v3.OrcaLoadReport);
+}
+
+message OrcaLoadReportRequest {
+  // Interval for generating Open RCA core metric responses.
+  google.protobuf.Duration report_interval = 1;
+  // Request costs to collect. If this is empty, all known requests costs tracked by
+  // the load reporting agent will be returned. This provides an opportunity for
+  // the client to selectively obtain a subset of tracked costs.
+  repeated string request_cost_names = 2;
+}
+```
+
+The definition for this service can be found at
+https://github.com/cncf/xds/blob/main/xds/service/orca/v3/orca.proto.
+
+Alternatively, an HTTP endpoint can be exposed by the load reporting server. A GET or HEAD request
+to this endpoint will return a response with the `endpoint-load-metrics` header that contains the
+load report in one of the formats listed above.
+
+# ORCA Integration Considerations
```

Comment:
I think this section also needs to be updated, since it no longer seems that relevant as currently written.  In general, I think users are not going to choose a sidecar or library model based solely on ORCA; instead, they will probably have one of those models already and will want to know what the options are for adding ORCA support.

I think what we should be explaining here is that there are cases where a smart data plane like Envoy or gRPC can handle ORCA for you, transparently to the application, and there are other cases where the application will need to generate the ORCA data directly.